client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -75,28 +70,29 @@ public ReaderGroupManagerImpl(String scope, Controller controller, AbstractClien
         this.controller = controller;
     }
 
-    private Stream createStreamHelper(String streamName, StreamConfiguration config) {
-        getAndHandleExceptions(controller.createStream(scope, streamName, StreamConfiguration.builder()
-                                                                          .scalingPolicy(config.getScalingPolicy())
-                                                                          .build()),
-                               RuntimeException::new);
-        return new StreamImpl(scope, streamName);
-    }
-
     @Override
     public void createReaderGroup(String groupName, ReaderGroupConfig config) {
         log.info("Creating reader group: {} for streams: {} with configuration: {}", groupName,
                 Arrays.toString(config.getStartingStreamCuts().keySet().toArray()), config);
         NameUtils.validateReaderGroupName(groupName);
-        createStreamHelper(getStreamForReaderGroup(groupName), StreamConfiguration.builder()
-                                                                                  .scalingPolicy(ScalingPolicy.fixed(1))
-                                                                                  .build());
-        ReaderGroupConfig ctrlConfig = getThrowingException(controller.createReaderGroup(scope, groupName, config));
+        if (config.getReaderGroupId() == ReaderGroupConfig.DEFAULT_UUID) {
```

Comment:
a general question - this RG id is on the builder. so does that mean user application can actually set the RG id themselves or do we make this method on config.builder package private so that only we can set the rg id?

Reply:
fixed.